account_fiscal_year/tests/test_account_fiscal_year.py

```diff
@@ -0,0 +1,132 @@
+from odoo import fields
+from odoo.tests import SavepointCase
+
+
+class TestFiscalPosition(SavepointCase):
+    def check_compute_fiscal_year(
+        self, company, date, expected_date_from, expected_date_to
+    ):
+        """Compute the fiscal year at a certain date for the company
+        passed as parameter.
+        Then, check if the result matches the
+        'expected_date_from'/'expected_date_to' dates.
+
+        :param company: The company.
+        :param date: The date belonging to the fiscal year.
+        :param expected_date_from: The expected date_from after computation.
+        :param expected_date_to: The expected date_to after computation.
+        """
+        current_date = fields.Date.from_string(date)
+        res = company.compute_fiscalyear_dates(current_date)
+        self.assertEqual(res["date_from"], fields.Date.from_string(expected_date_from))
+        self.assertEqual(res["date_to"], fields.Date.from_string(expected_date_to))
+
+    def test_default_fiscal_year(self):
+        """Basic case with a fiscal year xxxx-01-01 - xxxx-12-31."""
+        company = self.env.ref("base.main_company")
+        company.fiscalyear_last_day = 31
```

Comment:
Side note: It's strange that `fiscalyear_last_day` and `fiscalyear_last_mont` fields are still in [Odoo community](https://github.com/odoo/odoo/blob/d8254c130d4b04a0f1ff80990fe28e795bf9ad51/addons/account/models/company.py#L41-L42).

Reply:
yes there are pieces of fiscal year everywhere